clang/lib/Basic/SourceManager.cpp

```diff
@@ -1064,6 +1064,14 @@ bool SourceManager::isAtStartOfImmediateMacroExpansion(SourceLocation Loc,
   return true;
 }
 
+void SourceManager::setLocationForTimeTrace(SourceLocation Loc,
+                                            llvm::TimeTraceMetadata &M) const {
```

Comment:
Could you move outside of source manager?

The method can be implemented independent of source manager and there's no reason for source manager to depend on `TimeTraceMetadata`.
We might need a separate header for this, but that's ok.